src/test/java/cloud/fogbow/ras/core/plugins/interoperability/aws/network/v2/AwsV2NetworkPluginTest.java

```diff
@@ -0,0 +1,616 @@
+
+		Mockito.verify(client, Mockito.times(1)).describeSubnets(Mockito.any(DescribeSubnetsRequest.class));
+		Mockito.verify(client, Mockito.times(1)).describeRouteTables();
+
+		Assert.assertEquals(expected, instance);
+	}
+
+	// test case: When calling the isReady method with the cloud state AVAILABLE,
+	// this means that the state of network is READY and it must return true.
+	@Test
+	public void testIsReadySuccessful() {
+		// set up
+		String cloudState = AwsV2StateMapper.AVAILABLE_STATE;
+
+		// exercise
+		boolean status = this.plugin.isReady(cloudState);
+
+		// verify
+		Assert.assertTrue(status);
+	}
+
+	// test case: When calling the isReady method with the cloud states different
+	// than AVAILABLE, this means that the state of compute is not READY and it must
+	// return false.
+	@Test
+	public void testIsReadyUnsuccessful() {
+		// set up
+		String[] cloudStates = { ANY_VALUE, AwsV2StateMapper.PENDING_STATE, AwsV2StateMapper.UNKNOWN_TO_SDK_VERSION_STATE };
+
+		for (String cloudState : cloudStates) {
+			// exercise
+			boolean status = this.plugin.isReady(cloudState);
+
+			// verify
+			Assert.assertFalse(status);
+		}
+	}
+		
+	// test case: Whenever you call the hasFailed method, no matter the value, it
+	// must return false.
+	@Test
+	public void testHasFailed() {
+		// set up
+		String cloudState = ANY_VALUE;
+
+		// exercise
+		boolean status = this.plugin.hasFailed(cloudState);
+
+		// verify
+		Assert.assertFalse(status);
+	}
+	
+	// test case: When calling the getGatewayFromRouteTables method, and a route
+	// list with a gateway ID different of local, it must return a default
+	// destination CIDR block.
+	@Test
+	public void testGetGatewayFromRouteTablesSuccessful() {
+		// set up
+		String gatewayId = ANY_VALUE;
+		String expected = DEFAULT_CIDR_DESTINATION;
+
+		Route route = Route.builder()
+				.gatewayId(gatewayId)
+				.destinationCidrBlock(expected)
+				.build();
+
+		List<Route> routes = new ArrayList<>();
+		routes.add(route);
+
+		// exercise
+		String gateway = this.plugin.getGatewayFromRouteTables(routes);
+
+		// verify
+		Assert.assertEquals(expected, gateway);
+	}
+	
+	// test case: When calling the getGatewayFromRouteTables method, and a route
+	// list with a gateway ID equals to local, it must return a null value.
+	@Test
+	public void testGetGatewayFromRouteTablesUnsuccessful() {
+		// set up
+		String gatewayId = AwsV2NetworkPlugin.LOCAL_GATEWAY_DESTINATION;
+		String expected = null;
+
+		Route route = Route.builder()
+				.gatewayId(gatewayId)
+				.build();
+
+		List<Route> routes = new ArrayList<>();
+		routes.add(route);
+
+		// exercise
+		String gateway = this.plugin.getGatewayFromRouteTables(routes);
+
+		// verify
+		Assert.assertEquals(expected, gateway);
+	}
+	
+	// test case: When calling the getGroupIdBySubnet method, with a valid subnet ID
+	// and client, and a key tag different of groupId, the UnexpectedException will
```

Comment:
actually it should throw a FogbowExpcetion, once we know it was expected

Reply:
Okay, I'll fix the test case.